subsys/bluetooth/audio/has_client.c

```diff
@@ -82,11 +88,176 @@ static enum bt_has_capabilities get_capabilities(const struct has_inst *inst)
 	return caps;
 }
 
+union preset_record {
+	struct bt_has_preset_record record;
+	struct {
+		uint8_t index;
+		enum bt_has_properties properties;
+		char name[BT_HAS_PRESET_NAME_MAX + 1]; /* + 1 byte for null-terminator */
+	};
```

Comment:
Is there any advantage of having this struct internally and expose 
```c
struct bt_has_preset_record {
	uint8_t index;
	enum bt_has_properties properties;
	const char name[];
};
```
externally? 

Could we not use this struct in the public API as well?

Reply:
changed that